hydroflow_plus/src/stream.rs

```diff
@@ -690,6 +690,38 @@ pub(super) fn deserialize_bincode<T: DeserializeOwned>(tagged: bool) -> Pipeline
 }
 
 impl<'a, T, W, N: Location<'a>> Stream<T, W, NoTick, N> {
+    pub fn decouple_process<P2>(
+        self,
+        other: &Process<'a, P2>,
+    ) -> Stream<T, Unbounded, NoTick, Process<'a, P2>>
+    where
+        N: CanSend<'a, Process<'a, P2>, In<T> = T, Out<T> = T>,
+        T: Clone + Serialize + DeserializeOwned,
+    {
+        self.send_bincode::<Process<'a, P2>, T>(other)
+    }
+
+    pub fn decouple_cluster<C2>(
+        self,
+        other: &Cluster<'a, C2>,
+    ) -> Stream<N::Out<T>, Unbounded, NoTick, Cluster<'a, C2>>
+    where
+        N: CanSend<'a, Cluster<'a, C2>, In<T> = (u32, T)>,
```

Comment:
We should restrict `Out<T>` here to be `(u32, T)` as well, and then use `send_bincode_interleaved` so that we guarantee that we have the same type on the input and output (the return type should be `Stream<T, ...>`).